File: lib/workflow_delete/workflow_delete.ex
defmodule WorkflowDelete do
  @external_resource "docs/workflow_delete.md"
  @moduledoc TemporalSdk.Utils.exdoc!("docs/workflow_delete.md")

  @spec run() :: :ok | no_return()
  def run() do
    {:ok, %{workflow_execution: we}} =
      TemporalSdk.start_workflow(:cluster_1, "default", WorkflowTerminate.Workflow)

    # Sleep commands are used to mitigate Temporal server eventual consistency
    Process.sleep(500)

    IO.puts("""
    WF execution state before deletion:
      #{TemporalSdk.get_workflow_state(:cluster_1, we) |> inspect()}
    """)

    {:ok, %{}} = TemporalSdk.delete_workflow(:cluster_1, we)
    Process.sleep(500)

    IO.puts("""
    WF execution state after deletion:
      #{TemporalSdk.get_workflow_state(:cluster_1, we) |> inspect()}
    """)
  end
end

File: src/workflow_delete/workflow_delete.erl
-module(workflow_delete).

% elp:ignore W0012 W0040 E1599
-moduledoc {file, "../../docs/workflow_delete.md"}.

-export([
    run/0
]).

-spec run() -> ok | no_return().
run() ->
    workflow_eviction:maybe_attach_telemetry(),
    {ok, #{workflow_execution := WE}} = temporal_sdk:start_workflow(
        cluster_1, "default", workflow_terminate_workflow
    ),
    %% Sleep commands are used to mitigate Temporal server eventual consistency
    timer:sleep(500),
    io:fwrite("WF execution state before deletion:~n  ~kp~n~n", [
        temporal_sdk:get_workflow_state(cluster_1, WE)
    ]),
    {ok, #{}} = temporal_sdk:delete_workflow(cluster_1, WE),
    timer:sleep(500),
    io:fwrite("WF execution state after deletion:~n  ~kp~n~n", [
        temporal_sdk:get_workflow_state(cluster_1, WE)
    ]).
